docs/source/notes/contributing.rst

```diff
@@ -0,0 +1,313 @@
+
+.. tip:: `Live Server <https://github.com/ritwickdey/vscode-live-server>`_ is a great extension if you are looking for a static-files server to serve contents in ``docs/build/html``.
+
+Writing new documents
+^^^^^^^^^^^^^^^^^^^^^
+
+.. |link_example| raw:: html
+
+   <code class="docutils literal notranslate">`Link text &lt;https://domain.invalid/&gt;`_</code>
+
+.. |link_example_2| raw:: html
+
+   <code class="docutils literal notranslate">`Link text &lt;https://domain.invalid/&gt;`__</code>
+
+.. |githublink_example| raw:: html
+
+   <code class="docutils literal notranslate">:githublink:`path/to/file.ext`</code>
+
+.. |githublink_example_2| raw:: html
+
+   <code class="docutils literal notranslate">:githublink:`text &lt;path/to/file.ext&gt;`</code>
+
+`ReStructuredText <https://docutils.sourceforge.io/docs/user/rst/quickstart.html>`_ is our documentation language. Please find the reference of RST `here <https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html>`__.
+
+.. tip:: Sphinx has `an excellent cheatsheet of rst <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_ which contains almost everything you might need to know to write a elegant document.
+
+**Dealing with sections.** ``=`` for sections. ``-`` for subsections. ``^`` for subsubsections. ``"`` for paragraphs.
+
+**Dealing with images.** Images should be put into ``docs/img`` folder. Then, reference the image in the document with relative links. For example, ``.. image:: ../../img/example.png``.
+
+**Dealing with codes.** We recommend using ``.. code-block:: python`` to start a code block. The ``python`` here annotates the syntax highlighting.
+
+**Dealing with links.** Use |link_example| for inline web links. Note that use one underline might cause `"duplicated target name" error <https://stackoverflow.com/questions/27420317/restructured-text-rst-http-links-underscore-vs-use>`_ when multiple targets share the same name. In that case, use double-underline to avoid the error: |link_example_2|.
+
+Other than built-in directives provided by Sphinx, we also provide some custom directives:
+
+* ``.. cardlinkitem::``: A tutorial card, useful in :doc:`../tutorials`.
+* |githublink_example| or |githublink_example_2|: reference a file on the GitHub. Linked to the same commit id as where the documentation is built.
+
+Writing new tutorials
+^^^^^^^^^^^^^^^^^^^^^
+
+Our tutorials are powered by `sphinx-gallery <https://sphinx-gallery.github.io/>`. Sphinx-gallery is an extension that builds an HTML gallery of examples from any set of Python scripts.
+
+To contribute a new tutorial, here are the steps to follow:
+
+1. Create a notebook styled python file. If you want it executed while inserted into documentation, save the file under ``examples/tutorials/``. If your tutorial contains other auxiliary scripts which are not intended to be included into documentation, save them under ``examples/tutorials/scripts/``.
```

Comment:
better to add a reference link to examples/tutorials, so that users can easily find complete examples there.

Reply:
I don't think it's necessary for "developers". They can easily find them in their cloned repo. We don't need to point to a possibly-outdated github address.

Reply:
so you imply that we should no use github address anyway?

Reply:
We should use them when we want to guide customers to read the repo on GitHub website.